src/Substrait.Lineage/LineageExtractorRelVisitor.cs

```diff
@@ -0,0 +1,20 @@
+﻿using Substrait.Core;
+using Substrait.Relation;
+using ProtoRel = Substrait.Protobuf.Rel;
+
+namespace Substrait.Lineage
+{
+  /// <summary>
+  /// A concrete implementation of <see cref="SubstraitRelVisitor"/> for processing Substrait relations and extracting
+  /// lineage information.
+  /// </summary>
+  public class LineageExtractorRelVisitor : SubstraitRelVisitor
```

Comment:
Is this class really needed?  I would think the normal behavior for a system would be to do something like:

```
public void OnMessage(byte[] message)
{
  ProtoRel protoMessage = DecodeMessage(message);
  Rel root = new ProtoConverter().ToRel(protoMessage);
  HandleMessage(root);
}

private void HandleMessage(Rel rel)
{
  // Do some stuff
  LogQuery(rel);
  // Do some other stuff
}

private void LogQuery(Rel rel)
{
  new LoggingVisitor(logger).Visit(rel);
}
```

In other words, I would expect to convert at the boundaries, then visiting the converted plan as needed.  Instead this appears to convert each time I need to visit something.